### ISSUE TITLE
simplify fails to output records in sorted order if same node supports multiple edgesets with disjoint children

### ISSUE THREAD

**Issue by petrelharp** [jeromekelleher/msprime#182]
I have here an example on which simplify() fails at the
```
            assert(cr->node > last_node);
```
statement in `lib/tree_sequence.c::tree_sequence_load_records`. I realize this is a temporary interface, but the problem lies upstream in `simplify()`: by inserting print statements and running
```
ts = msprime.load_text(nodes=open("nodes.txt","r"), edgesets=open("edges.txt","r"))
ts.simplify()
```
I find that it is outputting these records (format: time: left, right - node):
```
3.000000 : 0.000000, 1000000.000000 - 10
     1,  2,  
7.000000 : 231101.870531, 1000000.000000 - 11
     4,  7,  
10.000000 : 0.000000, 231101.870531 - 12
     0,  7,  
10.000000 : 0.000000, 1000000.000000 - 13
     3,  6,  
12.000000 : 0.000000, 231101.870531 - 14
     4,  12,  
12.000000 : 0.000000, 1000000.000000 - 15
     10,  13,  
12.000000 : 31827.206804, 204879.104124 - 16
     5,  9,  
12.000000 : 231101.870531, 1000000.000000 - 14
     0,  11,  
python3: lib/tree_sequence.c:1000: tree_sequence_load_records: Assertion `cr->node > last_node' failed.
```
... which are indeed not in order.  I have winnowed down the files somewhat, but attempting to start from the records implicated by the breakpoints and working outwards to produce a MWE was unsuccessful.

**Comment by petrelharp** [comment]
The files are available at https://github.com/petrelharp/msprime/tree/debug_simplify/debug

**Comment by petrelharp** [comment]
Another one, outputting:
```
3.000000 : 662691.523585, 1000000.000000 - 6
     1,  2,  
9.000000 : 0.000000, 472008.946498 - 7
     0,  1,  
10.000000 : 472008.946498, 662691.523585 - 8
     0,  2,  
11.000000 : 195189.224782, 1000000.000000 - 9
     4,  5,  
14.000000 : 0.000000, 166255.775593 - 10
     4,  5,  
14.000000 : 483813.530235, 662691.523585 - 11
     1,  8,  
14.000000 : 662691.523585, 1000000.000000 - 11
     0,  6,  
23.000000 : 0.000000, 472008.946498 - 12
     2,  7,  
23.000000 : 166255.775593, 195189.224782 - 13
     4,  5,  
23.000000 : 472008.946498, 483813.530235 - 12
     1,  8,  
```

**Comment by petrelharp** [comment]
Looking at the two examples I made a test that gets the problem.  It doesn't show when running `test_topology.py`, but simplify outputs
```
1.000000 : 0.000000, 0.400000 - 6
     2,  3,  
1.000000 : 0.000000, 0.500000 - 7
     0,  1,  
1.000000 : 0.000000, 0.500000 - 8
     4,  5,  
1.000000 : 0.400000, 1.000000 - 9
     2,  3,  
1.000000 : 0.500000, 1.000000 - 7
     4,  5,  
python3: lib/tree_sequence.c:1000: tree_sequence_load_records: Assertion `cr->node > last_node' failed.
```

**Comment by jeromekelleher** [comment]
Thanks  a million for this @petrelharp, fantastic bug report! Fixed (hopefully) over at #183.
